ios/RNApplePayWrappers.m

```diff
@@ -0,0 +1,245 @@
+
+    for (NSDictionary *summaryItemDict in summaryItemsArray) {
+        JPPaymentSummaryItem *item = [self summaryItemFromDictionary:summaryItemDict];
+        [summaryItems addObject:item];
+    }
+
+    return summaryItems;
+}
+
++ (JPPaymentSummaryItem *)summaryItemFromDictionary:(NSDictionary *)dictionary {
+    NSString *label = dictionary[@"label"];
+    NSString *amountString = dictionary[@"amount"];
+    NSDecimalNumber *amount = [NSDecimalNumber decimalNumberWithString:amountString];
+    NSNumber *type = dictionary[@"type"];
+
+    return [JPPaymentSummaryItem itemWithLabel:label
+                                        amount:amount
+                                          type:type.intValue];
+}
+
++ (MerchantCapability)merchantCapabilitiesFromAppleConfiguration:(NSDictionary *)appleConfiguration {
+    NSNumber *selectedCapabilitiesNumber = appleConfiguration[@"merchantCapabilities"];
+    int selectedCapabilitiesValue = selectedCapabilitiesNumber.intValue;
+    MerchantCapability capabilities = 0;
+
+    if (selectedCapabilitiesValue & IOSApplePayCapabilityAll) {
+        return MerchantCapability3DS | MerchantCapabilityEMV | MerchantCapabilityCredit | MerchantCapabilityDebit;
+    }
+
+    if (selectedCapabilitiesValue & IOSApplePayCapability3DS) {
+        capabilities |= MerchantCapability3DS;
+    }
+
+    if (selectedCapabilitiesValue & IOSApplePayCapabilityEMV) {
+        capabilities |= MerchantCapabilityEMV;
+    }
+
+    if (selectedCapabilitiesValue & IOSApplePayCapabilityCredit) {
+        capabilities |= MerchantCapabilityCredit;
+    }
+
+    if (selectedCapabilitiesValue & IOSApplePayCapabilityDebit) {
+        capabilities |= MerchantCapabilityDebit;
+    }
+
+    return capabilities;
+}
+
++ (NSArray<PaymentShippingMethod *> *)shippingMethodsFromAppleConfiguration:(NSDictionary *)appleConfiguration {
+    NSArray *shippingMethodsArray = appleConfiguration[@"shippingMethods"];
+    NSMutableArray *mappedShippingMethods = [NSMutableArray new];
+
+    for (NSDictionary *shippingMethodDict in shippingMethodsArray) {
+        PaymentShippingMethod *method = [self shippingMethodFromDictionary:shippingMethodDict];
+        [mappedShippingMethods addObject:method];
+    }
+
+    return mappedShippingMethods;
+}
+
++ (PaymentShippingMethod *)shippingMethodFromDictionary:(NSDictionary *)dictionary {
+
+    NSString *identifier = dictionary[@"identifier"];
+    NSString *detail = dictionary[@"detail"];
+    NSString *label = dictionary[@"label"];
+    NSString *amountString = dictionary[@"amount"];
+    NSNumber *typeNumber = dictionary[@"type"];
+
+    return [[PaymentShippingMethod alloc] initWithIdentifier:identifier
+                                                      detail:detail
+                                                       label:label
+                                                      amount:[NSDecimalNumber decimalNumberWithString:amountString]
```

Comment:
this can throw `NSDecimalNumberOverflowException` or `NSDecimalNumberUnderflowException` or `NSDecimalNumberExactnessException` or `NSDecimalNumberDivideByZeroException` we need to wrap this call in a try / catch block

Reply:
Can this (and all other validations) be done as part of the Error Handling story?
Validating all required / optional parameters would take some time and prevent others from starting working on their tasks

Reply:
sure, let's do that as part of the Error Handling story